### PR TITLE
Feature/plot

### PR DESCRIPTION
**Pull request by v1kko**
New plotting functionality
```yaml
plot:
  data:
  - db: jet
    run: 251
    shot: 94875
    user: g2vazizi
  plots:
  - add_time_slider: false
    x: null
    xlabel: null
    y: profiles_1d/0/electrons/density_thermal
    ylabel: null
```
Allows you to specify multiple ids and multiple plots

Plotting over time (time_slider) does not yet work

### CODE REVIEW

**Comment by v1kko** [comment]
@stefsmeets , what do you think, is this the way to go?

File: duqtools/plot.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        kwargs
+    """
+
+    info('Extracting imas data')
+    # Gather all results and put them in a in-memory format
+    # (they should be small enough)
+    profiles = []
+    for entry in cfg().plot.data:
+        debug('Extracting database: %s, %s, %s, %s' %
+              (entry.db, entry.shot, entry.run, entry.user))
```
Comment (stefsmeets) [inline duqtools/plot.py:29]: I think you should be able to do this:

```suggestion
        debug('Extracting database: %s' % entry)
```

File: duqtools/plot.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        source = ImasLocation(db=entry.db,
+                              shot=entry.shot,
+                              run=entry.run,
+                              user=entry.user)
+        profiles.append(source.get_simple_IDS('core_profiles'))
```
Comment (stefsmeets) [inline duqtools/plot.py:34]: `entry` is already an `ImasLocation`, so this should work:

```suggestion
        profiles.append(entry.get_simple_IDS('core_profiles'))
```

File: duqtools/plot.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    for i, plot in enumerate(cfg().plot.plots):
+        info('Creating plot number %i' % i)
+        for profile in profiles:
+            y = profile.flat_fields[plot.y]
+            if plot.ylabel:
+                plt.ylabel(plot.ylabel)
+            else:
+                plt.ylabel(plot.y)
```
Comment (stefsmeets) [inline duqtools/plot.py:43]: Trying to hide some of the complexity here. Not sure if I like it. Maybe it's better using a method on `plot`, i.e. `plot.get_ylabel()`.

```suggestion
            plt.ylabel(plot.ylabel if plot.ylabel else plot.y)
```

File: duqtools/plot.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+            if plot.x:
+                x = profile.flat_fields[plot.x]
+            else:
+                x = np.linspace(0, 1, len(y))
+            if plot.xlabel:
+                plt.xlabel(plot.xlabel)
+            else:
+                plt.xlabel(plot.x)
```
Comment (stefsmeets) [inline duqtools/plot.py:52]:
```suggestion
            plt.xlabel(plot.xlabel if plot.xlabel else plot.x)
```

File: duqtools/plot.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            y = profile.flat_fields[plot.y]
+            if plot.ylabel:
+                plt.ylabel(plot.ylabel)
+            else:
+                plt.ylabel(plot.y)
+
+            if plot.x:
+                x = profile.flat_fields[plot.x]
+            else:
+                x = np.linspace(0, 1, len(y))
```
Comment (stefsmeets) [inline duqtools/plot.py:48]:
```suggestion
            x = profile.flat_fields[plot.x] if plot.x else np.linspace(0, 1, len(y))
```

**Review by stefsmeets — APPROVED** [review]
Cool, this worked right away when I tried it on Gateway! It's a good start to be able to generate some of the plots. We can wrap it in a GUI after. Let's see what kind of feedback we get first.

I put some comments, feel free to include them as you see fit.

File: duqtools/plot.py (hunk 3 above)
Comment (v1kko) [reply duqtools/plot.py]: I fixed it via pydantic, but I am not sure its better, what do you think? @stefsmeets

File: duqtools/plot.py (hunk 5 above)
Comment (v1kko) [reply duqtools/plot.py:37]: I'll  leave this one as is

File: duqtools/plot.py (hunk 3 above)
Comment (v1kko) [reply duqtools/plot.py]: It doesn't even work now, let me fix that

**Pull request merged**